### PR TITLE
New Resource: aws_cognito_user_pool_domain

### PR DESCRIPTION
**Pull request by radeksimko**
## Test results

```
TF_ACC=1 go test ./aws -v -run=TestAccAWSCognitoUserPoolDomain_ -timeout 120m
=== RUN   TestAccAWSCognitoUserPoolDomain_basic
--- PASS: TestAccAWSCognitoUserPoolDomain_basic (43.74s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	43.772s
```

### CODE REVIEW

File: aws/resource_aws_cognito_user_pool_domain.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCognitoUserPoolDomain() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCognitoUserPoolDomainCreate,
+		Read:   resourceAwsCognitoUserPoolDomainRead,
+		Delete: resourceAwsCognitoUserPoolDomainDelete,
+		Importer: &schema.ResourceImporter{
```
Comment (Ninir) [inline aws/resource_aws_cognito_user_pool_domain.go]: What do you think of adding a test for the import?

File: aws/resource_aws_cognito_user_pool_domain.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCognitoUserPoolDomainCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cognitoidpconn
+	log.Print("[DEBUG] Creating Cognito User Pool Domain")
```
Comment (Ninir) [inline aws/resource_aws_cognito_user_pool_domain.go]: Nitpick: I would add the params as `%#v` here moving this after the structure definition

File: aws/resource_aws_cognito_user_pool_domain.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	_, err := conn.CreateUserPoolDomain(params)
+	if err != nil {
+		return fmt.Errorf("Error creating Cognito User Pool Domain: %s", err)
+	}
+
+	d.SetId(domain)
+
+	stateConf := resource.StateChangeConf{
+		Pending: []string{
+			cognitoidentityprovider.DomainStatusTypeCreating,
```
Comment (Ninir) [inline aws/resource_aws_cognito_user_pool_domain.go]: 👍

File: aws/resource_aws_cognito_user_pool_domain.go [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+		return err
+	}
+
+	desc := domain.DomainDescription
+
+	d.Set("aws_account_id", desc.AWSAccountId)
+	d.Set("cloudfront_distribution_arn", desc.CloudFrontDistribution)
+	d.Set("s3_bucket", desc.S3Bucket)
+	d.Set("user_pool_id", desc.UserPoolId)
+	d.Set("version", desc.Version)
```
Comment (Ninir) [inline aws/resource_aws_cognito_user_pool_domain.go]: Nitpick: What do you think of checking that these attributes are actually set, in the related test?

**Comment by radeksimko** [comment]
Addressed all comments, but I'll ship it after `1.3.1` as discussed earlier.

File: aws/resource_aws_cognito_user_pool_domain.go [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+		Delete: resourceAwsCognitoUserPoolDomainDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"domain": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
```
Comment (Ninir) [inline aws/resource_aws_cognito_user_pool_domain.go]: What do you think of adding validation for the domain also, as in http://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_CreateUserPoolDomain.html#API_CreateUserPoolDomain_RequestSyntax?

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.7.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#170-january-12-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**